### ISSUE TITLE
Chat + vid_restart crash

### ISSUE THREAD

**Issue by kharnov** [Unvanquished/Unvanquished#825]
```
[H] kharnov: yhelllagaldf 
[H] kharnov: hlskfjhlkjgh 
]/vid_restart 
]
Program received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fffc0da1700 (LWP 4774)]
std::_Rb_tree<Rocket::Core::StringBase<char>, std::pair<Rocket::Core::StringBase<char> const, std::vector<Rocket::Core::EventDispatcher::Listener, std::allocator<Rocket::Core::EventDispatcher::Listener> > >, std::_Select1st<std::pair<Rocket::Core::StringBase<char> const, std::vector<Rocket::Core::EventDispatcher::Listener, std::allocator<Rocket::Core::EventDispatcher::Listener> > > >, std::less<Rocket::Core::StringBase<char> >, std::allocator<std::pair<Rocket::Core::StringBase<char> const, std::vector<Rocket::Core::EventDispatcher::Listener, std::allocator<Rocket::Core::EventDispatcher::Listener> > > > >::_M_lower_bound (
    this=0x7fffb9af5378, __k=..., __y=0x7fffb9af5380, __x=0x7fff40400000) at /usr/include/c++/5/bits/stl_tree.h:1628
1628		if (!_M_impl._M_key_compare(_S_key(__x), __k))
(gdb) bt
#0  std::_Rb_tree<Rocket::Core::StringBase<char>, std::pair<Rocket::Core::StringBase<char> const, std::vector<Rocket::Core::EventDispatcher::Listener, std::allocator<Rocket::Core::EventDispatcher::Listener> > >, std::_Select1st<std::pair<Rocket::Core::StringBase<char> const, std::vector<Rocket::Core::EventDispatcher::Listener, std::allocator<Rocket::Core::EventDispatcher::Listener> > > >, std::less<Rocket::Core::StringBase<char> >, std::allocator<std::pair<Rocket::Core::StringBase<char> const, std::vector<Rocket::Core::EventDispatcher::Listener, std::allocator<Rocket::Core::EventDispatcher::Listener> > > > >::_M_lower_bound
    (this=0x7fffb9af5378, __k=..., __y=0x7fffb9af5380, __x=0x7fff40400000) at /usr/include/c++/5/bits/stl_tree.h:1628
#1  std::_Rb_tree<Rocket::Core::StringBase<char>, std::pair<Rocket::Core::StringBase<char> const, std::vector<Rocket::Core::EventDispatcher::Listener, std::allocator<Rocket::Core::EventDispatcher::Listener> > >, std::_Select1st<std::pair<Rocket::Core::StringBase<char> const, std::vector<Rocket::Core::EventDispatcher::Listener, std::allocator<Rocket::Core::EventDispatcher::Listener> > > >, std::less<Rocket::Core::StringBase<char> >, std::allocator<std::pair<Rocket::Core::StringBase<char> const, std::vector<Rocket::Core::EventDispatcher::Listener, std::allocator<Rocket::Core::EventDispatcher::Listener> > > > >::find (
    this=this@entry=0x7fffb9af5378, __k=...) at /usr/include/c++/5/bits/stl_tree.h:2295
#2  0x00007fffb6468eaf in std::map<Rocket::Core::StringBase<char>, std::vector<Rocket::Core::EventDispatcher::Listener, std::allocator<Rocket::Core::EventDispatcher::Listener> >, std::less<Rocket::Core::StringBase<char> >, std::allocator<std::pair<Rocket::Core::StringBase<char> const, std::vector<Rocket::Core::EventDispatcher::Listener, std::allocator<Rocket::Core::EventDispatcher::Listener> > > > >::find (__x=..., 
    this=0x7fffb9af5378) at /usr/include/c++/5/bits/stl_map.h:846
#3  Rocket::Core::EventDispatcher::DetachEvent (this=0x7fffb9af5370, type=..., listener=listener@entry=0x7fffb9b42f10, in_capture_phase=in_capture_phase@entry=false)
    at /home/kharnov/Unvanquished/libs/libRocket/Source/Core/EventDispatcher.cpp:75
#4  0x00007fffb63d3e2f in Rocket::Core::Element::RemoveEventListener (this=<optimized out>, event=..., listener=listener@entry=0x7fffb9b42f10, in_capture_phase=in_capture_phase@entry=false)
    at /home/kharnov/Unvanquished/libs/libRocket/Source/Core/Element.cpp:1047
#5  0x00007fffb6334386 in RocketChatField::OnChildRemove (this=0x7fffb9b42a60, child=<optimized out>) at /home/kharnov/Unvanquished/src/cgame/rocket/rocketChatField.h:63
#6  0x00007fffb63d6cc3 in Rocket::Core::Element::~Element (this=0x7fffb9e53430, __in_chrg=<optimized out>) at /home/kharnov/Unvanquished/libs/libRocket/Source/Core/Element.cpp:127
#7  0x00007fffb64cd973 in Rocket::Core::Lua::LuaDocument::~LuaDocument (this=0x7fffb9e53430, __in_chrg=<optimized out>) at /home/kharnov/Unvanquished/libs/libRocket/Source/Core/Lua/LuaDocument.h:39
#8  Rocket::Core::Lua::LuaDocument::~LuaDocument (this=0x7fffb9e53430, __in_chrg=<optimized out>) at /home/kharnov/Unvanquished/libs/libRocket/Source/Core/Lua/LuaDocument.h:39
#9  0x00007fffb63d6bd8 in Rocket::Core::Element::OnReferenceDeactivate (this=<optimized out>) at /home/kharnov/Unvanquished/libs/libRocket/Source/Core/Element.cpp:1686
#10 0x00007fffb6436e62 in Rocket::Core::Lua::LuaType<Rocket::Core::ElementDocument>::gc_T (L=0x7fffb93faeb0) at /home/kharnov/Unvanquished/libs/libRocket/Include/Rocket/Core/Lua/LuaType.inl:185
#11 0x00007fffc1c4c92d in ?? () from /usr/lib/x86_64-linux-gnu/liblua5.2.so.0
#12 0x00007fffc1c4cc62 in ?? () from /usr/lib/x86_64-linux-gnu/liblua5.2.so.0
#13 0x00007fffc1c4c2af in ?? () from /usr/lib/x86_64-linux-gnu/liblua5.2.so.0
#14 0x00007fffc1c4ceb1 in ?? () from /usr/lib/x86_64-linux-gnu/liblua5.2.so.0
#15 0x00007fffc1c4e13f in ?? () from /usr/lib/x86_64-linux-gnu/liblua5.2.so.0
#16 0x00007fffc1c4f62e in ?? () from /usr/lib/x86_64-linux-gnu/liblua5.2.so.0
#17 0x00007fffc1c54e1e in ?? () from /usr/lib/x86_64-linux-gnu/liblua5.2.so.0
#18 0x00007fffb632fee7 in Rocket_Shutdown () at /home/kharnov/Unvanquished/src/cgame/rocket/rocket.cpp:410
#19 0x00007fffb62f2b3e in CG_Shutdown () at /home/kharnov/Unvanquished/src/cgame/cg_main.cpp:1765
#20 0x00007fffb62c8d35 in VM::<lambda()>::operator() (__closure=<optimized out>) at /home/kharnov/Unvanquished/src/cgame/cg_api.cpp:71
#21 Util::apply_impl<VM::VMHandleSyscall(uint32_t, Util::Reader)::<lambda()>, std::tuple<> > (func=<optimized out>, tuple=<optimized out>) at /home/kharnov/Unvanquished/daemon/src/common/Util.h:103
#22 Util::apply<VM::VMHandleSyscall(uint32_t, Util::Reader)::<lambda()>, std::tuple<> > (func=<optimized out>, tuple=<optimized out>) at /home/kharnov/Unvanquished/daemon/src/common/Util.h:108
#23 IPC::detail::HandleMsg<VM::VMHandleSyscall(uint32_t, Util::Reader)::<lambda()>, IPC::Message<IPC::Id<0u, 2u> >, IPC::Reply<> > (channel=..., func=<optimized out>, reader=...)
    at /home/kharnov/Unvanquished/daemon/src/common/IPC/Channel.h:217
#24 IPC::HandleMsg<IPC::SyncMessage<IPC::Message<IPC::Id<0u, 2u> > >, VM::VMHandleSyscall(uint32_t, Util::Reader)::<lambda()> > (channel=..., func=<optimized out>, 
    reader=<error reading variable: access outside bounds of object referenced via synthetic pointer>) at /home/kharnov/Unvanquished/daemon/src/common/IPC/Channel.h:241
#25 VM::VMHandleSyscall (id=<optimized out>, reader=...) at /home/kharnov/Unvanquished/src/cgame/cg_api.cpp:72
#26 0x00007fffb637f3af in CommonInit (rootSocket=<optimized out>) at /home/kharnov/Unvanquished/daemon/src/shared/VMMain.cpp:66
#27 0x00007fffb637f64b in vmMain (rootSocket=<optimized out>) at /home/kharnov/Unvanquished/daemon/src/shared/VMMain.cpp:104
#28 0x00000000005981af in VM::<lambda()>::operator() (__closure=0x51ec5a8) at /home/kharnov/Unvanquished/daemon/src/engine/framework/VirtualMachine.cpp:352
#29 std::_Bind_simple<VM::CreateInProcessNativeVM(std::pair<IPC::Socket, IPC::Socket>, Str::StringRef, VM::VMBase::InProcessInfo&)::<lambda()>()>::_M_invoke<> (this=0x51ec5a8)
    at /usr/include/c++/5/functional:1531
#30 std::_Bind_simple<VM::CreateInProcessNativeVM(std::pair<IPC::Socket, IPC::Socket>, Str::StringRef, VM::VMBase::InProcessInfo&)::<lambda()>()>::operator() (this=0x51ec5a8)
    at /usr/include/c++/5/functional:1520
#31 std::thread::_Impl<std::_Bind_simple<VM::CreateInProcessNativeVM(std::pair<IPC::Socket, IPC::Socket>, Str::StringRef, VM::VMBase::InProcessInfo&)::<lambda()>()> >::_M_run(void) (this=0x51ec590)
    at /usr/include/c++/5/thread:115
#32 0x00007ffff4656030 in ?? () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#33 0x00007ffff3e676aa in start_thread (arg=0x7fffc0da1700) at pthread_create.c:333
#34 0x00007ffff3b9ceed in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:109 
```

**Comment by Viech** [comment]
@kharnov, reason for closing?

**Comment by Kangz** [comment]
I think it was fixed in 79fc559f268f277e98636e058ca558a239cb9e63
